include/zephyr/bluetooth/bluetooth.h

```diff
@@ -268,7 +268,7 @@ bool bt_is_ready(void);
  *
  * @sa @kconfig{CONFIG_BT_DEVICE_NAME_MAX}.
  *
- * @param name New name
+ * @param name New name, must be null terminated
```

Comment:
I'm not going to oppose this, but it feels a bit superfluous. We also don't have such a documentation convention elsewhere (AFAIK) like here: https://github.com/zephyrproject-rtos/zephyr/blob/4e201f21c8a7f58698e68a5f1cca72433f9778ac/include/zephyr/kernel.h#L1293

If an API takes a string parameter, and doesn't have any additional parameters that could be used to indicate the length, then the string parameter can be assumed to be a normal C string, i.e. null-terminated.

Reply:
I agree with you and that was my understanding of the API too. But the bug is a proof that it need to be stated.

The C strings are not trivial, we should always be careful when manipulating them. We may be used to them but not everyone is and they are definitely not the norm among the programming language. So I think being explicit in the documentation is important.

Reply:
I would rather document this in a generic way: When `char *` is used we should always expect/assume that it is a NULL terminated string. This should not be an API or BT specific thing, but a Zephyr-wise assumption/expectation IMO

Reply:
The risk with Zephyr-wise assumption is that some people miss it... At least here in the documentation, if someone use the function it will be remained of the expectation.

Reply:
> The risk with Zephyr-wise assumption is that some people miss it...

Wouldn't the same people miss it for the `strxxx` functions like `strlen` as well?

Reply:
Probably, yes. But is that a reason for not trying to make life easier on our side? 😅

@jhedberg and you seems to be against that so if you really want me to remove it, I will do it